### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by shaurya-src**

### CODE REVIEW

File: Assignment-4/contact_app/lib/screens/contacts/contacts.dart [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    getPermissions();
+  }
+
+  getPermissions() async {
+    if (await Permission.contacts.request().isGranted) {
+      getAllContacts();
+    }
+  }
+
+  getAllContacts() async {
```
Comment (Akanksha-nuclei) [inline Assignment-4/contact_app/lib/screens/contacts/contacts.dart:32]: put this in view model

File: Assignment-4/contact_app/lib/screens/contacts/contacts.dart [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    List<Contact> _contacts = (await ContactsService.getContacts()).toList();
+    setState(() {
+      contactList = _contacts;
+    });
+  }
+
+  searchContacts() async {
```
Comment (Akanksha-nuclei) [inline Assignment-4/contact_app/lib/screens/contacts/contacts.dart:39]: segregate the view and view model and helper function

**Review by Akanksha-nuclei — COMMENTED** [review]
request changes

File: Assignment-4/contact_app/lib/screens/contacts/contacts.dart (hunk 1 above)
Comment (shaurya-src) [reply Assignment-4/contact_app/lib/screens/contacts/contacts.dart]: Done

File: Assignment-4/contact_app/lib/screens/contacts/contacts.dart (hunk 2 above)
Comment (shaurya-src) [reply Assignment-4/contact_app/lib/screens/contacts/contacts.dart]: Done

**Pull request merged**